code/modules/client/preference_setup/general/04_equipment.dm

```diff
@@ -48,70 +48,83 @@
 
 /datum/category_item/player_setup_item/general/equipment/gather_save_parameters()
 	return list(
-		"underwear" = pref.underwear,
-		"undershirt" = pref.undershirt,
-		"socks" = pref.socks,
+		"all_underwear" = pref.all_underwear,
+		"all_underwear_metadata" = pref.all_underwear_metadata,
 		"backbag" = pref.backbag,
 		"backbag_style" = pref.backbag_style,
 		"id" = pref.current_character,
 		"ckey" = PREF_CLIENT_CKEY
 	)
 
 /datum/category_item/player_setup_item/general/equipment/sanitize_character(var/sql_load = 0)
+	if(!istype(pref.all_underwear))

```

Comment:
What's not being done:
In case of `sql_load == 1`, you're not deserializing JSON into DM lists.

It should look roughly like this:
```dm
if (sql_load)
  if (pref.all_underwear)
    pref.all_underewear = json_decode(pref.all_underwear)
  if (pref.all_underwear_metadata)
    pref.all_underwear_metadata = json_decode(pref.all_underwear_metadata)
```

Reply:
You'll also want to wrap both of these json_decodes into a try/catch, and replace the values with `null` in the catch.